accepted/0000-yomi-initial-integration.md

```diff
@@ -0,0 +1,361 @@
+- Feature Name: Initial integration with Yomi (MVP)
+- Start Date: 2019-10-08
+- RFC PR: (leave this empty)
+
+# Summary
+[summary]: #summary
+
+Yomi (Yet one more installer) is an installer for openSUSE (Leap,
+Tumbleweed, MicroOS) and SLE (WIP) expressed as a set of Salt
+states. This RFC propose the integration of Yomi for provisioning of
+new systems in Uyuni.
+
+# Motivation
+[motivation]: #motivation
+
+Currently in Uyuni / SUMA we have several ways for installing a new
+system:
+
+- For RedHat based systems we can use the Web UI to create, update and
+  manage Kickstart files. Those files can be designed to be applied
+  for different nodes. (IS THIS TRUE?)
+
+- For [open]SUSE based systems we can upload an AutoYaST file, but
+  this is usually designed for a single type on system (IS THIS TRUE?)
+
+- We can leverage Cobbler manage the PXE-Boot server and automate the
+  installation of new systems (via Kickstart, AutoYaST and Cheetah or
+  Jinja2 template engine)
+
+- SUMA Retailer allows the installation of new systems dumping a
+  partition image (generated by KIWI, for example) into the local root
+  partition.
+
+For one side we can see some unbalance between the Kickstart and the
+AutoYaST integration: the Kickstart template edition is more naturally
+integrated into the UI.
+
+For other side we are moving Uyuni forward into a better and more
+consistent integration with SaltStack, but only the retail version
+(based on images) is using this configuration management system.
+
+To improve this situation we propose the integration of the Yomi
+installer into Uyuni. Yomi re-formulate the installation process as a
+set of Salt states. In this model the pillar will contain the
+definition of:
+
+- Partitions and file systems used in the node
+
+- Bootloader parameters
+
+- Repositories and packages installed
+
+- Services enabled and disabled
+
+- Users and certificates
+
+As we can see, this proposal is (1) moving closer the features of
+Kickstart based systems into the SUSE family, and (2) using SaltStack
+to orchestrate the new installation.
+
+# Detailed design
+[design]: #detailed-design
+
+## Integrating Yomi as a Salt state
+
+The core of the proposal is to add Yomi into the catalog of formulas
+available in Uyuni.
+
+As Yomi is indeed a Salt formula, this integration can be done
+directly, installing the `yomi-formula` package into the Manager.
+
+The user can later decide to add the Yomi state into the `top.sls` for
+this system, fill the pillar data using the SaltForm provided in the
+package, and launch the installer by applying the highstate.
```

Comment:
I assume by `SaltForm` you mean `form.yml` or Formula form, right?

Note that as soon as formula is enabled for the registered system it will be automatically applied in the highstate (see `mgr_master_tops.py`) so no need for user to manually modify `top.sls`.

Reply:
> I assume by `SaltForm` you mean `form.yml` or Formula form, right?

Correct. Is there better name to reference this feature? I can update the RFC to use the correct names.

> Note that as soon as formula is enabled for the registered system it will be automatically applied in the highstate (see `mgr_master_tops.py`) so no need for user to manually modify `top.sls`.

I see. I think that this problem is documented later. Maybe makes sense to have a different workflow for this case, and if the grains that is mentioned later is present, skip the automatic `state.highstate`?

Reply:
Absolutely. How this works is that when formula is enabled for the minion, then is is written to `/srv/susemanager/formula_data/minion_formulas.json` under minion id name. This file is that parsed by external pillar [suma_minion.py#L147](https://github.com/uyuni-project/uyuni/blob/master/susemanager-utils/susemanager-sls/modules/pillar/suma_minion.py#L147) into the minion pillar. Manager tops module provide access to `/usr/share/susemanager/formulas/states/` directory where resides [formulas.sls](https://github.com/uyuni-project/uyuni/blob/master/susemanager-utils/susemanager-sls/formulas/states/formulas.sls) which include all the formulas from this pillar. So anywhere in this chain can be added a jinja processing to selectively disable formula from highstate.
Another question however is if we wanted to do that from overall formulas strategy. For our image sync formula we solved that that our image sync state by default does nothing, only if enabled in formula it will include and apply actual state (see [image-synchronize/init.sls](https://github.com/uyuni-project/retail/blob/master/image-sync-formula/image-synchronize/init.sls))

Reply:
(Resolve this conversation, as some time ago I changed the mechanics of detection: https://github.com/openSUSE/yomi/commit/ccc1e406e066b6331adf3a512372903d59019f19, so not need of any external grain)